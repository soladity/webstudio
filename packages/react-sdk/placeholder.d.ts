declare module "__CLIENT__" {
  import type { FontAsset, ImageAsset, System } from "@webstudio-is/sdk";

  export const siteName: string;

  export const favIconAsset: ImageAsset | undefined;

  export const socialImageAsset: ImageAsset | undefined;

  // Font assets on current page (can be preloaded)
  export const pageFontAssets: FontAsset[];

  export const pageBackgroundImageAssets: ImageAsset[];

  export const CustomCode: () => ReactNode;

  export const Page: (props: { system: System }) => ReactNode;
}

declare module "__SERVER__" {
  import type { PageMeta, System } from "@webstudio-is/sdk";

  export const loadResources: (props: {
    system: System;
  }) => Promise<Record<string, unknown>>;

  export const getPageMeta: (props: {
    system: System;
    resources: Record<string, any>;
  }) => PageMeta;

  export const formsProperties: Map<
    string,
    { method?: string; action?: string }
  >;

  type Params = Record<string, string | undefined>;
  export const getRemixParams: ({ ...params }: Params) => Params;

  export const projectId: string;

  export const contactEmail: undefined | string;
}

declare module "__SITEMAP__" {
  export const sitemap: Array<{
    path: string;
    lastModified: string;
  }>;
}
