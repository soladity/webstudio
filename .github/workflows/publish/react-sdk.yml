name: SDK publish
on:
  push:
    paths:
      - "packages/react-sdk/**/*"
      - ".github/workflows/publish/react-sdk.yml"
      - ".github/actions/publish/action.yaml"
jobs:
  publish:
    runs-on: ubuntu-latest
    steps:
      - uses: actions/checkout@v3
      - uses: ./.github/actions/publish
        with:
          package: "./packages/react-sdk/package.json"
          token: ${{ secrets.NPM_TOKEN }}
