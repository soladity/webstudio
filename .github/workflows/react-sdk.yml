name: React SDK
on:
  push:
    paths:
      - "packages/react-sdk/**/*"
      - ".github/workflows/react-sdk.yml"
      - ".github/actions/publish-test/action.yaml"
    branches:    
      - main
jobs:
  publish:
    runs-on: ubuntu-latest
    steps:
      - uses: actions/checkout@v3
      - uses: ./.github/actions/publish
        with:
          workdir: ./packages/react-sdk
          filter: "@webstudio-is/react-sdk"
          token: ${{ secrets.NPM_TOKEN }}
