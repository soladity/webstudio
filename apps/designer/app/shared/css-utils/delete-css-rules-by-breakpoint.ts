import type { Instance } from "@webstudio-is/react-sdk";
import type { Breakpoint } from "@webstudio-is/css-data";

export const deleteCssRulesByBreakpoint = (
  instance: Instance,
  breakpointId: Breakpoint["id"]
) => {
  const cssRules = [...instance.cssRules];
  cssRules.forEach((cssRule, index) => {
    if (cssRule.breakpoint === breakpointId) {
      instance.cssRules.splice(index, 1);
    }
  });

  for (const child of instance.children) {
    if (typeof child === "string") {
      continue;
    }
    deleteCssRulesByBreakpoint(child, breakpointId);
  }
};
