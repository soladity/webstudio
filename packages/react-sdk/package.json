{
  "name": "@webstudio-is/react-sdk",
  "version": "0.75.0",
  "description": "Webstudio JavaScript / TypeScript API",
  "author": "Webstudio <github@webstudio.is>",
  "homepage": "https://webstudio.is",
  "type": "module",
  "scripts": {
    "dev": "build-package --watch",
    "build": "build-package",
    "build:args": "generate-arg-types './src/components/*.tsx ./src/app/custom-components/*.tsx !./src/**/*.stories.tsx !./src/**/*.ws.tsx' && prettier --write \"**/*.props.ts\"",
    "dts": "tsc --project tsconfig.dts.json",
    "typecheck": "tsc --noEmit --emitDeclarationOnly false",
    "test": "NODE_OPTIONS=--experimental-vm-modules jest --passWithNoTests",
    "lint": "eslint ./src --ext .ts,.tsx --max-warnings 0",
    "checks": "pnpm typecheck && pnpm lint && pnpm test"
  },
  "devDependencies": {
    "@jest/globals": "^29.3.1",
    "@remix-run/react": "^1.15.0",
    "@types/react": "^18.0.35",
    "@types/react-dom": "^18.0.11",
    "@webstudio-is/jest-config": "workspace:^",
    "@webstudio-is/scripts": "workspace:^",
    "@webstudio-is/tsconfig": "workspace:^",
    "jest": "^29.3.1",
    "react": "^18.2.0",
    "react-dom": "^18.2.0",
    "type-fest": "^3.7.1",
    "typescript": "5.1.3",
    "zod": "^3.21.4"
  },
  "peerDependencies": {
    "@remix-run/react": "^1.15.0",
    "react": "^18.2.0",
    "react-dom": "^18.2.0",
    "zod": "^3.19.1"
  },
  "dependencies": {
    "@nanostores/react": "^0.4.1",
    "@webstudio-is/asset-uploader": "workspace:^",
    "@webstudio-is/css-data": "workspace:^",
    "@webstudio-is/css-engine": "workspace:^",
    "@webstudio-is/fonts": "workspace:^",
    "@webstudio-is/generate-arg-types": "workspace:^",
    "@webstudio-is/project-build": "workspace:^",
    "detect-font": "^0.1.5",
    "html-tags": "^3.2.0",
    "nanoevents": "^7.0.1",
    "nanoid": "^3.3.6",
    "nanostores": "^0.7.1"
  },
  "exports": {
    ".": {
      "source": "./src/index.ts",
      "types": "./lib/types/index.d.ts",
      "import": "./lib/index.js",
      "require": "./lib/cjs/index.js"
    },
    "./css-normalize": {
      "source": "./src/css/normalize.ts",
      "types": "./lib/types/css/normalize.d.ts",
      "import": "./lib/css/normalize.js",
      "require": "./lib/cjs/css/normalize.js"
    }
  },
  "files": [
    "lib/*",
    "src/*",
    "!*.test.*"
  ],
  "license": "MIT",
  "private": false,
  "sideEffects": false
}
