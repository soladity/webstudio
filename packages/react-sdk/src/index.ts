export * from "./css";
export * from "./tree";
export * from "./pubsub";
export * from "./app";
export * from "./components/components-utils";
export {
  type WsComponentPropsMeta,
  type WsComponentMeta,
  type ComponentState,
  type PresetStyle,
  componentCategories,
  stateCategories,
  defaultStates,
} from "./components/component-meta";
export * from "./embed-template";
export {
  usePropUrl,
  usePropAsset,
  getInstanceIdFromComponentProps,
} from "./props";
export { type Params, ReactSdkContext } from "./context";
