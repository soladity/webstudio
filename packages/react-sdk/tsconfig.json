{
  // Extends can't use node's resolution rn https://github.com/dividab/tsconfig-paths/issues/209#issuecomment-1177261091
  "extends": "../../node_modules/@webstudio-is/tsconfig/base.json",
  "include": [
    "./src",
    "./src/**/__generated__/*.json",
    "../../@types/**/*.d.ts"
  ],
  "compilerOptions": {
    "lib": ["DOM", "DOM.Iterable", "ES2020"],
    "module": "ES2020",
    "jsx": "react-jsx",
    "target": "ES2020",
    "strict": true,
    "baseUrl": ".",
    "outDir": "./lib",
    "rootDir": "./src"
  }
}
