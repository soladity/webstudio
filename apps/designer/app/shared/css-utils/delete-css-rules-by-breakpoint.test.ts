import { describe, test, expect } from "@jest/globals";
import { type Instance } from "@webstudio-is/react-sdk";
import { deleteCssRulesByBreakpoint } from "./delete-css-rules-by-breakpoint";

describe("Delete breakpoint", () => {
  test("basic", () => {
    const instance: Instance = {
      type: "instance",
      component: "Box",
      id: "1",
      cssRules: [{ breakpoint: "delete this", style: {} }],
      children: [
        {
          type: "instance",
          component: "Box",
          id: "2",
          cssRules: [{ breakpoint: "do not delete", style: {} }],
          children: [
            {
              type: "instance",
              component: "Box",
              id: "3",
              cssRules: [{ breakpoint: "delete this", style: {} }],
              children: [],
            },
          ],
        },
      ],
    };

    deleteCssRulesByBreakpoint(instance, "delete this");
    expect(instance).toMatchSnapshot();
  });
});
