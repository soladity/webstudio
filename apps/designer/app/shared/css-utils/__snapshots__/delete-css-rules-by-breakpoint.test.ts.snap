// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Delete breakpoint basic 1`] = `
{
  "children": [
    {
      "children": [
        {
          "children": [],
          "component": "Box",
          "cssRules": [],
          "id": "3",
          "type": "instance",
        },
      ],
      "component": "Box",
      "cssRules": [
        {
          "breakpoint": "do not delete",
          "style": {},
        },
      ],
      "id": "2",
      "type": "instance",
    },
  ],
  "component": "Box",
  "cssRules": [],
  "id": "1",
  "type": "instance",
}
`;
